gr-qtgui/lib/fosphor_formatter_impl.cc

```diff
@@ -0,0 +1,286 @@
+
+        hv = (hv - d) * e + d;
+        hv = std::max(std::min(1.0f, hv), 0.0f);
+    }
+};
+
+
+} // namespace
+
+fosphor_formatter::sptr fosphor_formatter::make(int fft_size,
+                                                int num_bins,
+                                                int input_decim,
+                                                int waterfall_decim,
+                                                int histo_decim,
+                                                double scale,
+                                                double alpha,
+                                                double epsilon,
+                                                double trise,
+                                                double tdecay)
+{
+    return gnuradio::make_block_sptr<fosphor_formatter_impl>(fft_size,
+                                                             num_bins,
+                                                             input_decim,
+                                                             waterfall_decim,
+                                                             histo_decim,
+                                                             scale,
+                                                             alpha,
+                                                             epsilon,
+                                                             trise,
+                                                             tdecay);
+}
+
+
+fosphor_formatter_impl::fosphor_formatter_impl(int fft_size,
+                                               int num_bins,
+                                               int input_decim,
+                                               int waterfall_decim,
+                                               int histo_decim,
+                                               double scale,
+                                               double alpha,
+                                               double epsilon,
+                                               double trise,
+                                               double tdecay)
+    : hier_block2("fosphor_formatter",
+                  io_signature::make(1, 1, sizeof(gr_complex)),
+                  io_signature::make(2, 2, sizeof(unsigned char) * fft_size)),
+      // Init sub-blocks
+      d_s2v(gr::blocks::stream_to_vector::make(sizeof(gr_complex), fft_size)),
+      d_input_decim(
+          gr::blocks::keep_one_in_n::make(sizeof(gr_complex) * fft_size, input_decim)),
+      d_fft(gr::fft::fft_v<gr_complex, true /* forward */>::make(
+          fft_size, gr::fft::window::blackman_harris(fft_size), true /* shift */)),
+      d_c2m(gr::blocks::complex_to_mag_squared::make(fft_size)),
+      d_log(gr::blocks::nlog10_ff::make(calculate_log10_scale(scale),
+                                        fft_size,
+                                        20 * std::log10(static_cast<float>(fft_size)))),
```

Comment:
This may be part of the problem. I think the intent is to divide by N^2, but this is multiplying by N^2. Even after that, I still haven't been able to find a scaling that matches the fosphor visualization against the GR frequency sink or Gqrx.

Changing the sign on "k" here and setting "scale" to 6 puts tones at 0dB at all FFT sizes. However, wideband signals and the noise floor don't match other the frequency sink or Gqrx.